client-tools/Shell/RequestFuzzer.cpp

```diff
@@ -0,0 +1,278 @@
+  while (true) {
+    BodyOperation bodyOp = static_cast<BodyOperation>(
+        _randContext.mt() % BodyOperation::MAX_BODY_OP_VALUE);
+    if (_recursionDepth > _maxDepth && bodyOp <= BodyOperation::ADD_OBJECT) {
+      continue;
+    }
+    switch (bodyOp) {
+      case ADD_ARRAY: {
+        builder.openArray(_randContext.mt() % 2 ? true : false);
+        uint32_t numMembers = _randContext.mt() % _arrayNumMembers;
+        for (uint32_t i = 0; i < numMembers; ++i) {
+          ++_recursionDepth;
+          randomizeBodyInternal(builder);
+          --_recursionDepth;
+        }
+        builder.close();
+        break;
+      }
+      case ADD_OBJECT: {
+        builder.openObject(_randContext.mt() % 2 ? true : false);
+        uint32_t numMembers = _randContext.mt() % _objNumMembers;
+        if (_tempObjectKeys.size() <= _recursionDepth) {
+          _tempObjectKeys.resize(_recursionDepth + 1);
+        } else {
+          _tempObjectKeys[_recursionDepth].clear();
+        }
+        ++_recursionDepth;
+        for (uint32_t i = 0; i < numMembers; ++i) {
+          while (true) {
+            _tempStr.clear();
+            generateRandReadableAsciiString(_tempStr);
+            if (_tempObjectKeys[_recursionDepth - 1].emplace(_tempStr).second) {
+              break;
+            }
+          }
+          // key
+          builder.add(velocypack::Value(_tempStr));
+          // value
+          randomizeBodyInternal(builder);
+        }
+        --_recursionDepth;
+        _tempObjectKeys[_recursionDepth].clear();
+        builder.close();
+        break;
+      }
+      case ADD_CHAR_SEQ: {
+        _tempStr.clear();
+        generateRandReadableAsciiString(_tempStr);
+        builder.add(velocypack::Value(_tempStr));
+        break;
+      }
+      default:
+        TRI_ASSERT(false);
+    }
+    break;
+  }
+}
+
+template<typename T>
+T RequestFuzzer::generateRandNumWithinRange(T min, T max) {
+  return min + (_randContext.mt() % (max + 1 - min));
+}
+
+int32_t RequestFuzzer::generateRandInt32() {
+  uint32_t uintValue = _randContext.mt();
+  int32_t intValue;
+  memcpy(&intValue, &uintValue, sizeof(uintValue));
+  return intValue;
+}
+
+void RequestFuzzer::generateRandAsciiChar(std::string& input) {
+  input[generateRandNumWithinRange<uint32_t>(0, input.size())] =
+      generateRandNumWithinRange<uint32_t>(0x0, 0x7F);
+}
+
+void RequestFuzzer::generateRandReadableAsciiString(std::string& input) {
+  static constexpr char chars[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+
+  uint32_t randStrLength = generateRandNumWithinRange<uint32_t>(
+      0, _randContext.maxRandAsciiStringLength);
+
+  for (uint32_t i = 0; i < randStrLength; ++i) {
+    input += chars[_randContext.mt() % (sizeof(chars) - 1)];
+  }
+}
+
+void RequestFuzzer::logStart() {
+  LOG_TOPIC("871a6", INFO, arangodb::Logger::COMMUNICATION)
+      << "Started fuzzing... Seed: " << _seed;
+}
```

Comment:
I think we don't need a log message. Would be better if the fuzzer at the end returned the effectively used seed as part of its return value.